### DISCUSSION TITLE
copying/syncing of pg_hba.conf file between servers

### DISCUSSION THREAD

**Discussion by jerrykan** [hapostgres/pg_auto_failover#986] [Q&A] [answered]
I've been testing out pg_auto_failover and it seem really promissing. However during testing I've noticed some unexpected behaviour with the `pg_hba.conf` files.

When initially setting up the second node with `pg_autoctl create postgres --skip-pg-hba ...` it seems that the `pg_hba.conf` file from the first node is copied across to the second node.

Also once both nodes are up an running and a `pg_autoctl perform switchover` is run it seems that the `pg_hba.conf` file on the secondary (being promoted to primary) is copied over to the primary (being demoted to a secondary).

Both of these seem at odds with the documentation that - as far as I can tell - states that if the `--skip-pg-hba` option is used, then the `pg_autoctl` won't make any changes to the `pg_hba.conf` file and it should be manually managed by an admin.

This leads to a couple of questions:
  - Is the "syncing" of the `pg_hba.conf` files during init/failover expected behaviour? is there any way to disable it?
  - Why is the `pg_hba.conf` from the secondary server synced between servers instead of the version on the primary? It would seem that the `pg_hba.conf` file on the primary would be the most up-to-date and correct file (ie. because the primary has been correctly accepting connections), while the `pg_hba.conf` could potential contain a misconfiguration.

**Comment by DimCitus** [comment, marked as answer]
Hi @jerrykan,

When doing secondary provisioning and then later failover/switchover pg_auto_failover relies on Postgres core tooling, namely `pg_basebackup` and `pg_rewind`. Postgres core tools typically sync the whole content of the PGDATA directory, so that includes `pg_hba.conf`. The usual way to deal with that situation with Postgres is to use the same HBA rules on the primary and standby nodes. After all, as you're mentioning, the standby/secondary nodes are deployed for only one reason: take over the primary role when needed. For them to be ready, best scenario is that the HBA is kept in sync with the primary.

Hope this clarifies the situation. Closing the discussion, please consider asking more questions if needed though.

**Reply by jerrykan** [reply]
Hello @DimCitus thanks for the info. The behaviour makes a bit more sense now.
